src/simulation/LoadGenerator.cpp

```diff
@@ -1662,54 +1669,36 @@ std::pair<SorobanResources, uint32_t>
-    resources.instructions =
-        sampleDiscrete(cfg.LOADGEN_INSTRUCTIONS_FOR_TESTING,
-                       cfg.LOADGEN_INSTRUCTIONS_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_INSTRUCTIONS);
 
-    // Respect both contract size limit and TX write byte
-    // limits including key overhead
-    uint32_t const keyOverhead = 40;
+    // Sample a random Wasm size
     uint32_t wasmSize = sampleDiscrete(
         cfg.LOADGEN_WASM_BYTES_FOR_TESTING,
         cfg.LOADGEN_WASM_BYTES_DISTRIBUTION_FOR_TESTING, DEFAULT_WASM_BYTES);
-    resources.readBytes =
-        sampleDiscrete(cfg.LOADGEN_IO_KILOBYTES_FOR_TESTING,
-                       cfg.LOADGEN_IO_KILOBYTES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_IO_KILOBYTES) *
-        1024;
-    // Allocate enough write bytes to write the whole Wasm plus the 40 bytes of
-    // the key, plus extra drawn from the distribution.
-    resources.writeBytes =
-        wasmSize + keyOverhead +
-        sampleDiscrete(cfg.LOADGEN_IO_KILOBYTES_FOR_TESTING,
-                       cfg.LOADGEN_IO_KILOBYTES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_IO_KILOBYTES) *
-            1024;
 
-    auto writeKeys = LedgerTestUtils::generateUniqueValidSorobanLedgerEntryKeys(
-        sampleDiscrete(cfg.LOADGEN_NUM_DATA_ENTRIES_FOR_TESTING,
-                       cfg.LOADGEN_NUM_DATA_ENTRIES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_NUM_DATA_ENTRIES));
-
-    for (auto const& key : writeKeys)
-    {
-        resources.footprint.readWrite.emplace_back(key);
-    }
-
-    auto readKeys = LedgerTestUtils::generateUniqueValidSorobanLedgerEntryKeys(
-        sampleDiscrete(cfg.LOADGEN_NUM_DATA_ENTRIES_FOR_TESTING,
-                       cfg.LOADGEN_NUM_DATA_ENTRIES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_NUM_DATA_ENTRIES));
+    // Estimate VM instantiation cost, with some additional buffer to increase
+    // the chance that this instruction count is sufficient.
+    uint32_t constexpr vmConstTerm = 451'626;
+    uint64_t constexpr vmLinearTerm = 45405;
+    uint32_t constexpr vmShiftTerm = 7;
+    uint32_t constexpr vmBufferPerByte = 100;
+    // Perform multiplication as uint64_t to avoid overflow
+    uint32_t linearResult = static_cast<uint32_t>(
+        (vmLinearTerm * static_cast<uint64_t>(wasmSize)) >> vmShiftTerm);
+    resources.instructions =
+        vmConstTerm + linearResult + (vmBufferPerByte * wasmSize);
 
-    for (auto const& key : readKeys)
-    {
-        resources.footprint.readOnly.emplace_back(key);
-    }
+    // Allocate enough write bytes to write the whole Wasm plus the 40 bytes of
+    // the key with some additional buffer to increase the chance that this
+    // write size is sufficient
+    uint32_t constexpr keyOverhead = 40;
+    uint32_t constexpr writeBuffer = 16;
+    resources.writeBytes = wasmSize + keyOverhead + writeBuffer;
+
+    // Need a single read/write key for the Wasm
+    auto writeKeys =
```

Comment:
I don't think generating a random key is correct here, as we populate the footprint with the proper key later in `LoadGenerator::sorobanRandomWasmTransaction`.

Reply:
Good catch. I removed the key generation in 98d1ca84b359685dbb164bacaa4c63e628c92df6